src/spikeinterface/generation/session_displacement_generator.py

```diff
@@ -0,0 +1,144 @@
+from spikeinterface.generation.drifting_generator import (
+    generate_probe,
+    fix_generate_templates_kwargs,
+    calculate_displacement_unit_factor,
+)
+from spikeinterface.core.generate import (
+    generate_unit_locations,
+    generate_sorting,
+    generate_templates,
+)
+import numpy as np
+from spikeinterface.generation.noise_tools import generate_noise
+from spikeinterface.core.generate import setup_inject_templates_recording
+from spikeinterface.core import InjectTemplatesRecording
+
+
+def generate_inter_session_displacement_recordings(
+    num_units=250,
+    rec_durations=(10, 10, 10),
+    rec_shifts=((0, 0), (0, 25), (0, 50)),
+    non_rigid_gradient=None,
+    sampling_frequency=30000.0,
+    probe_name="Neuropixel-128",
+    generate_probe_kwargs=None,
+    generate_unit_locations_kwargs=dict(
```

Comment:
TODO: these are direct copied from `generate_drifting_recording`. Maybe these can both use a shared function e.g. as in #3226

Reply:
Will proceed with #3226 after this PR